annotation/tests/test_tasks_services.py

```diff
@@ -0,0 +1,311 @@
+        True,
+        {1},
+        {2},
+        {3},
+        True,
+        True,
+        "Cannot finish validation task. There are not processed pages: " "{3}",
+    ),
+)
+
+
+@pytest.fixture(scope="function")
+def db_session():
+    return Mock(spec=Session)
+
+
+@pytest.fixture(scope="function")
+def task_info():
+    return {
+        "is_validation": True,
+        "user_id": 1,
+        "job_id": 2,
+    }
+
+
+@pytest.mark.parametrize(
+    ("validation_type", "is_validation"),
+    (
+        (ValidationSchema.validation_only, True),
+        ("other_type", False),
+        ("other_type", True),
+    ),
+)
+def test_validate_task_info_positive_case(
+    db_session: Session,
+    task_info: dict,
+    validation_type: ValidationSchema,
+    is_validation: bool,
+):
+    task_info["is_validation"] = is_validation
+
+    with patch(
+        "annotation.tasks.services.validate_users_info"
+    ) as mock_validate_users_info, patch(
+        "annotation.tasks.services.validate_files_info"
+    ) as mock_validate_files_info:
+        validate_task_info(db_session, task_info, validation_type)
+        mock_validate_users_info.assert_called_once_with(
+            db_session, task_info, validation_type
+        )
+        mock_validate_files_info.assert_called_once_with(db_session, task_info)
+
+
+@pytest.mark.parametrize(
+    ("validation_type", "is_validation"),
+    ((ValidationSchema.validation_only, False),),
+)
+def test_validate_task_info_negative_case(
+    db_session: Session,
+    task_info: dict,
+    validation_type: ValidationSchema,
+    is_validation: bool,
+):
+    task_info["is_validation"] = is_validation
+
+    with pytest.raises(FieldConstraintError):
+        validate_task_info(db_session, task_info, validation_type)
+
+
+@pytest.mark.parametrize(
+    "validation_type, is_validation, mock_function, mock_return_value",
+    (
+        (ValidationSchema.cross, True, "check_cross_annotating_pages", True),
+        (ValidationSchema.validation_only, True, None, True),
+        (ValidationSchema.validation_only, False, None, True),
+    ),
+)
+def test_validate_users_info_positive_case(
+    db_session: Session,
+    task_info: dict,
+    validation_type: ValidationSchema,
+    is_validation: bool,
+    mock_function: str,
+    mock_return_value: bool,
+):
+    task_info["is_validation"] = is_validation
+
+    if mock_function:
+        with patch(f"annotation.tasks.services.{mock_function}") as mock_func:
+            validate_users_info(db_session, task_info, validation_type)
+            mock_func.assert_called_once_with(db_session, task_info)
+    else:
+        db_session.query().filter_by().first.return_value = mock_return_value
+        validate_users_info(db_session, task_info, validation_type)
+
+
+@pytest.mark.parametrize(
+    ("is_validation", "expected_error_message"),
+    (
+        (True, "user 1 is not assigned as validator for job 2"),
+        (False, "user 1 is not assigned as annotator for job 2"),
```

Comment:
Use f"{task_info.job_id}" rather than hard-coding the error message. It would make the code more readable.